pkg/nsx/services/mediator/mediator.go

```diff
@@ -75,3 +79,18 @@ func (serviceMediator *ServiceMediator) GetPortsOfSubnet(nsxSubnetID string) (po
 	subnetPortList := serviceMediator.SubnetPortStore.GetByIndex(common.IndexKeySubnetID, nsxSubnetID)
 	return subnetPortList
 }
+
+func (serviceMediator *ServiceMediator) GetNodeIDByName(nodeName string) (string, error) {
+	nodes := serviceMediator.NodeStore.GetByIndex(common.IndexKeyNodeName, nodeName)
+	if len(nodes) == 0 {
+		return "", fmt.Errorf("node %s not found", nodeName)
+	}
+	if len(nodes) > 1 {
+		var nodeIDs []string
+		for _, node := range nodes {
+			nodeIDs = append(nodeIDs, *node.Id)
+		}
+		return "", fmt.Errorf("multiple node IDs found for node %s: %v", nodeName, nodeIDs)
+	}
+	return *nodes[0].Id, nil
```

Comment:
in NCP, it uses node['node_deployment_info']['external_id'], is that the same value with node.Id?

Reply:
Thanks for the info, will check it.

Reply:
The returned value of API `HostTransPortNodesClient.List` doesn't has external_id field, this is its full info:
```
{
    "node": {
        "Links": null,
        "Schema": null,
        "Self": null,
        "Revision": 0,
        "CreateTime": 1692781260902,
        "CreateUser": "system",
        "LastModifiedTime": 1692781260902,
        "LastModifiedUser": "system",
        "Protection": "NOT_PROTECTED",
        "SystemOwned": false,
        "Description": null,
        "DisplayName": "10.218.129.174",
        "Id": "2767b129-d5ee-4ce4-9aa9-c344b9c57abb",
        "ResourceType": "HostTransportNode",
        "Tags": null,
        "OriginSiteId": null,
        "OwnerId": "c27d08fe-85a3-455b-9688-3a60986a9867",
        "ParentPath": "/infra/sites/default/enforcement-points/default",
        "Path": "/infra/sites/default/enforcement-points/default/host-transport-nodes/2767b129-d5ee-4ce4-9aa9-c344b9c57abb",
        "RealizationId": "2767b129-d5ee-4ce4-9aa9-c344b9c57abb",
        "RelativePath": "2767b129-d5ee-4ce4-9aa9-c344b9c57abb",
        "RemotePath": "",
        "UniqueId": "2767b129-d5ee-4ce4-9aa9-c344b9c57abb",
        "Children": null,
        "MarkedForDelete": false,
        "Overridden": false,
        "DiscoveredNodeIdForCreate": null,
        "HostSwitchSpec": {},
        "IsOverridden": false,
        "MaintenanceMode": "DISABLED",
        "NodeDeploymentInfo": {
            "ComputeCollectionId": "82d31f2d-8bac-4f3b-ba3c-a8d78a441b24:domain-c9",
            "DiscoveredIpAddresses": [
                "178.177.121.26",
                "fe80::250:56ff:fe66:d432",
                "192.222.123.3",
                "fe80::250:56ff:fe68:dc61",
                "10.218.129.174",
                "fe80::425:26ff:fed9:9e5e",
                "fd01:0:106:304:425:26ff:fed9:9e5e",
                "169.254.1.1",
                "fe80::250:56ff:fe6d:f323"
            ],
            "DiscoveredNodeId": "82d31f2d-8bac-4f3b-ba3c-a8d78a441b24:host-12",
            "Dpu": null,
            "Fqdn": "sc1-10-218-129-174.xxx.xxx.xxx.com",
            "HostCredential": null,
            "IpAddresses": [
                "10.218.129.174"
            ],
            "MaintenanceModeState": null,
            "ManagedByServer": "10.218.129.18",
            "OsType": "ESXI",
            "OsVersion": "8.0.0",
            "WindowsInstallLocation": null
        }
    }
}
```
It looks that the values of `Id`, `RealizationId`, `UniqueId` are the same, and I compared it with nsxlib API:
```
{'_create_time': 1692781260902,
 '_create_user': 'system',
 '_last_modified_time': 1692781260902,
 '_last_modified_user': 'system',
 '_protection': 'NOT_PROTECTED',
 '_revision': 0,
 '_system_owned': False,
 'description': '',
 'display_name': '10.218.129.174',
 'host_switch_spec': {'host_switches': [{'cpu_config': [],
                                         'host_switch_id': '50 1c 15 50 f4 24 '
                                                           '0e 87-23 47 a4 84 '
                                                           '40 63 d9 d9',
                                         'host_switch_mode': 'STANDARD',
                                         'host_switch_name': 'wcp_vds_1',
                                         'host_switch_profile_ids': [{'key': 'UplinkHostSwitchProfile',
                                                                      'value': 'e6142768-1fdc-4ef1-8ad2-8e723d7dec2f'},
                                                                     {'key': 'VtepHAHostSwitchProfile',
                                                                      'value': '0de8282e-7385-4e8e-a905-0c11960db728'}],
                                         'host_switch_type': 'VDS',
                                         'ip_assignment_spec': {'ip_pool_id': 'b025cc56-0ba1-41db-a049-3a79e9b39dd1',
                                                                'resource_type': 'StaticIpPoolSpec'},
                                         'is_migrate_pnics': False,
                                         'not_ready': False,
                                         'transport_zone_endpoints': [{'transport_zone_id': '668e6d10-7c95-448d-a644-41a5a4f3ba26',
                                                                       'transport_zone_profile_ids': [{'profile_id': '52035bb3-ab02-4a08-9884-18631312e50a',
                                                                                                       'resource_type': 'BfdHealthMonitoringProfile'}]}],
                                         'uplinks': [{'uplink_name': 'uplink1',
                                                      'vds_uplink_name': 'uplink1'}]}],
                      'resource_type': 'StandardHostSwitchSpec'},
 'id': '2767b129-d5ee-4ce4-9aa9-c344b9c57abb',
 'is_overridden': False,
 'maintenance_mode': 'DISABLED',
 'node_deployment_info': {'compute_collection_id': '82d31f2d-8bac-4f3b-ba3c-a8d78a441b24:domain-c9',
                          'discovered_ip_addresses': ['178.177.121.26',
                                                      'fe80::250:56ff:fe66:d432',
                                                      '192.222.123.3',
                                                      'fe80::250:56ff:fe68:dc61',
                                                      '10.218.129.174',
                                                      'fe80::425:26ff:fed9:9e5e',
                                                      'fd01:0:106:304:425:26ff:fed9:9e5e',
                                                      '169.254.1.1',
                                                      'fe80::250:56ff:fe6d:f323'],
                          'discovered_node_id': '82d31f2d-8bac-4f3b-ba3c-a8d78a441b24:host-12',
                          'external_id': '2767b129-d5ee-4ce4-9aa9-c344b9c57abb',
                          'fqdn': 'sc1-10-218-129-174.xxx.xxx.xxx.com',
                          'id': '2767b129-d5ee-4ce4-9aa9-c344b9c57abb',
                          'ip_addresses': ['10.218.129.174'],
                          'managed_by_server': '10.218.129.18',
                          'os_type': 'ESXI',
                          'os_version': '8.0.0',
                          'resource_type': 'HostNode',
                          'windows_install_location': ''},
 'node_id': '2767b129-d5ee-4ce4-9aa9-c344b9c57abb',
 'resource_type': 'TransportNode',
 'tags': [{'scope': 'policyPath',
           'tag': '/infra/sites/default/enforcement-points/default/host-transport-nodes/2767b129-d5ee-4ce4-9aa9-c344b9c57abb'}]}
```
We can see previous value (`2767b129-d5ee-4ce4-9aa9-c344b9c57abb`) equals to `node['node_deployment_info']['external_id']`, so I chose `node.Id`.